### PR TITLE
ARGO-122 mulitenancy support for metric profiles

### PR DESCRIPTION
**Pull request by makistsantekidis**
* Renamed PoemProfiles to MetricProfiles

 * Added test for the List operation

 * Added support for the new schema of metricProfiles

### CODE REVIEW

File: app/metricProfiles/metricProfilesModel.go [hunk 1]
```diff
@@ -24,16 +24,29 @@
...
+	Services []Service `bson:"services" xml:"services"`
+}
+
+//Service struct to represent services with their metrics
+type Service struct {
+	Service string   `bson:"service" xml:"service,attr"`
+	Metrics []string `bson:"metrics" xml:"metrics"`
+}
+
+//----------------------------
```
Comment (pkoro) [inline app/metricProfiles/metricProfilesModel.go]: Add description here

**Comment by makistsantekidis** [comment]
please review
/cc @pkoro @dpavlos

File: app/metricProfiles/metricProfilesController.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	err := error(nil)
+	contentType := "text/xml"
+	charset := "utf-8"
+
+	//STANDARD DECLARATIONS END
+
+	tenantDbConfig, err := authentication.AuthenticateTenant(r.Header, cfg)
+
+	if err != nil {
+		code = http.StatusInternalServerError
```
Comment (dpavlos) [inline app/metricProfiles/metricProfilesController.go]: `AuthenticateTenant` returns two "types" of error. One for DB connection and one for authentication issues. However the final http code is always "500". Maybe you should handle this more appropriately. Same goes for the other functions below.

Comment (makistsantekidis) [inline app/metricProfiles/metricProfilesController.go]: there was some casual conversation about this and the consensus was to keep doing it that way for now until multi-tenancy is complete and then focus on creating some comprehensive error handling and logging to address the issue. So for now we should add it in the backlog and fix it later

File: app/metricProfiles/metricProfilesController.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	"net/http"
+	"strings"
+
+	"github.com/argoeu/argo-web-api/utils/authentication"
+	"github.com/argoeu/argo-web-api/utils/config"
+	"github.com/argoeu/argo-web-api/utils/mongo"
+	"github.com/gorilla/mux"
+)
+
+func ListPoems(r *http.Request, cfg config.Config) (int, http.Header, []byte, error) {
```
Comment (dpavlos) [inline app/metricProfiles/metricProfilesController.go]: Lacks `AuthenticateTenant` check.

File: app/metricProfiles/metricProfiles_test.go [hunk 4]
```diff
@@ -0,0 +1,576 @@
...
+						"org.sam.SRM-GetTURLs",
+						"org.sam.SRM-Ls",
+						"org.sam.SRM-LsDir",
+						"org.sam.SRM-Put"},
+				},
+			},
+		})
+
+	request, _ := http.NewRequest("GET", "/api/v1/poems", strings.NewReader(""))
+	request.Header.Set("x-api-key", suite.clientkey)
```
Comment (dpavlos) [inline app/metricProfiles/metricProfiles_test.go]: If `PoemList` is used for compatibility reasons, you don't need this. If this is the case, ignore the comment on line 41 in `metricProfilesController`.

Comment (makistsantekidis) [inline app/metricProfiles/metricProfiles_test.go]: Removing it now...

**Comment by dpavlos** [comment]
+1

**Pull request merged**